src/shogun/mathematics/graph/Tensor.h

```diff
@@ -171,10 +107,11 @@ namespace shogun
 				                  SGVector<typename Container::Scalar>,
 				                  Container>)
 				{
+					const auto cast_type = from<typename Container::Scalar>();
 					if (m_shape.size() > 1)
 						error("Tried to cast a multidimensional Tensor to a "
 						      "SGVector.");
-					if (from<typename Container::Scalar>() != m_type)
+					if (*cast_type != *m_type)
```

Comment:
i think std:shared_ptr<> != shared_ptr works... or? what i mean that are being called on the underlying stored pointer (the operator)...

Reply:
hmm it was failing on my machine, but I think it might have been gdb tricking me

Reply:
ah, it's because I wasn't using the static version from ngraph. Anyway, fixed it now